tarorpc/taro.proto

```diff
@@ -99,6 +99,13 @@ service Taro {
     proof file information the receiver needs to fully receive the asset.
     */
     rpc SendAsset (SendAssetRequest) returns (SendAssetResponse);
+
+    /*
+    SubscribeSendAssetEventNtfns registers a subscription to the event
+    notification stream which relates to the asset sending process.
+    */
+    rpc SubscribeSendAssetEventNtfns (SubscribeSendAssetEventNtfnsRequest)
```

Comment:
Zooming out a bit to think about more generically useful calls for future pplication developers, what if we generalized this to a sort of proof distribution listening event? AFAICT, we care about this rn, as we want to be able to write itests against instances where we receive, update, or send proofs. Perhaps we can morph this then into a call centered around notifying applications when we receive or sync new proofs? 

This would then generalize to the case of base universes. Then using this, an app can show a sort of loading bar when a user is starting new proof bootstrap

Reply:
I agree that it makes sense to ad notifications for receiving assets also. You mention proof distribution events, i think it makes sense to include other asset sending/receiving events also (like waiting for tx confirmation). Or should that sort of notification stream be restricted to proofs?

Reply:
Was starting to write a document to enumerate the entities within the daemon and the associated states but then I had a thought.  Might we want to, at least eventually, use this event-information conduit to ferry external events back into the daemon?  E.g. the daemon being notified about the state of a channel or HTLC from whatever the payment channel daemon (lnd) is doing.

Reply:
Currently, in some instances, external events do reach the taro daemon via notification channels. Here's an example where we receive a notification once an on-chain transaction has confirmed:
https://github.com/ffranr/taro/blob/d73b1f85c2ffc7f3b5e08afb2f40452e84788be4/tarofreighter/chain_porter.go#L269-L269

> Might we want to, at least eventually, use this event-information conduit to ferry external events back into the daemon?

I would suggest that the taro daemon should receive notification events from lnd in a similar fashion to how on-chain related events are received.

Reply:
Currently, in some instances, external events do reach the taro daemon via notification channels. Here's an example where we receive a notification once an on-chain transaction has confirmed:
https://github.com/ffranr/taro/blob/d73b1f85c2ffc7f3b5e08afb2f40452e84788be4/tarofreighter/chain_porter.go#L269-L269

> Might we want to, at least eventually, use this event-information conduit to ferry external events back into the daemon?

I would suggest that the taro daemon should receive notification events from lnd in a similar fashion to how on-chain related events are received.